filesystem/unix_permission.go

```diff
@@ -11,8 +11,85 @@ import (
 	"syscall"
 )
 
+// whose permission should exist (for this file's permission) = condition (... file permission... Owner is root, Owner Can Read, Non Owner Cannot read
+// condition = string condition.? or OwnerIsRoot(uint) or OwnerOfFileIsRoot(filename).
+// whose permission should not exist
+// There are all rules.
+
+// somewhere else, get permission Mode
+// create rule: root is owner (condition. bitwise evaluator? or filename. filename!
+
+// OwnerRoot confition => call CheckFileOwnerRights
+// OwnerReadPermission => get Mode() => wrapper function like OwnerRead..
+//
+
 // CheckFileRights check that the given file path has been protected by the owner.
 // If the owner is changed, they need at least the sudo permission to override the owner.
+
+type FilePermission string
+
+const (
+	OwnerWrite  FilePermission = "OwnerWrite"
+	GroupWrite  FilePermission = "GroupWrite"
+	AnyoneWrite FilePermission = "AnyoneWrite"
+	OwnerRead   FilePermission = "OwnerRead"
+	AnyoneRead  FilePermission = "AnyoneRead"
+)
+
+var FilePermissionInHex = map[FilePermission]uint32{
+	OwnerWrite:  syscall.S_IWUSR,
+	GroupWrite:  syscall.S_IWGRP,
+	AnyoneWrite: syscall.S_IWOTH,
+	OwnerRead:   syscall.S_IRUSR,
+	AnyoneRead:  syscall.S_IROTH,
+}
```

Comment:
This is reinventing the wheel. os.File has a `Stat` struct which already expresses the file permissions.
```go
func main() {
	f, _ := os.Open("foo")
	st, _ := f.Stat()
	fmt.Printf("%v", st.Mode())
}
```

Running this prints `-rw-r--r--`

Reply:
This is to convert what the test writer want to check to what we compare with the actual file permission with. Getting that file stat is what this gets compared with. Not the same thing. And I'm not using that syscall thing in the testers because it's harder to understand / os specific stuff can be isolated here

Reply:
st.Mode <-> compare with this